homework/index.js

```diff
@@ -30,18 +15,178 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function createHeader(root) {
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    return header;
+  }
+
+  function createOptions(repositoryList, header) {
+    const select = createAndAppend('select', header, {
+      id: 'selector',
+    });
+    for (let i = 0; i < repositoryList.length; i++) {
+      const repository = repositoryList[i];
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: i,
+      });
+    }
+  }
+
+  function createMainContainer(root) {
+    const mainContainer = createAndAppend('div', root, {
+      id: 'container',
+    });
+    return mainContainer;
+  }
+
+  function containerListContributor(mainContainer) {
+    const leftContainer = createAndAppend('div', mainContainer, {
+      class: 'left-block frame',
+    });
+    const table = createAndAppend('table', leftContainer);
+    const listContributor = createAndAppend('tbody', table);
+    return listContributor;
+  }
+
+  function createContributorContainer(mainContainer) {
+    const rightContainer = createAndAppend('div', mainContainer, {
+      class: 'right-block frame',
+    });
+    createAndAppend('p', rightContainer, {
+      text: 'Contributions',
+      class: 'contributor',
+    });
+
+    return createAndAppend('ul', rightContainer, {
+      class: 'list1',
```

Comment:
Avoid names with a number suffix, unless there is a clear reason to do so. In this case it looks like you didn't bother to find a good name.

Reply:
The main problem I am not good in naming the things and secondly when I am trying something I used something sort to do quick sometimes i am using that one in many places and not to change for it to cause new problems, sometimes I am not taking it enough important. But none of those can not be excuse the excuse for that. I will try to improve my self more. I learn a lot from your reviews thanks..